src/main/java/com/itrex/java/lab/controller/ReporterController.java

```diff
@@ -0,0 +1,45 @@
+package com.itrex.java.lab.controller;
+
+import com.itrex.java.lab.entity.dto.CustomerReportDTO;
+import com.itrex.java.lab.entity.dto.OfferReportDTO;
+import com.itrex.java.lab.service.ReporterService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@RestController
+@RequestMapping("/report")
```

Comment:
Does this URL ("/report") contain only one or more reports?
If it contains several reports, it makes sense to rename "/report" as "/reports"

Reply:
But endpoints return only one report. I don't think that path name is so important.

Reply:
but you have two reports that can bring back this request mapping:
- /customers
- /offers
it is not one